rs/bitcoin/kyt/src/state.rs

```diff
@@ -0,0 +1,179 @@
+use crate::GetTxError;
+use bitcoin::{Address, Transaction};
+use ic_btc_interface::Txid;
+use std::cell::RefCell;
+use std::collections::BTreeMap;
+
+#[derive(Debug, Clone)]
+pub enum FetchTxStatus<T> {
+    PendingOutcall,
+    PendingRetry { buffer_size: u32 },
+    Error(GetTxError),
+    Fetched(FetchedTx<T>),
+}
+
+#[derive(Debug, Clone)]
+pub struct FetchedTx<T> {
+    pub tx: T,
+    /// The index is the index of the input of tx, and the value is
+    /// the calculated address when the input has been fetched.
+    pub input_addresses: Vec<Option<Address>>,
+}
+
+// Max number of concurrent http outcalls.
+const MAX_CONCURRENT: u32 = 100;
```

Comment:
The networking team says that about 50 outcalls can be sustained in parallel (per second).

Reply:
Thanks. Good to know! Will change this

Reply:
Good question.   :slightly_smiling_face: 
Yes, the size of the subnet matters. The 50 outcalls per second is their recommended bound for a subnet with 28 nodes. Since it's not a very tight bound, we can use the same bound for slightly larger subnets (31 or 34 replicas).